### PR TITLE
Add package READMEs and QUALITY_DECLARATION files

### PR DESCRIPTION
**Pull request by brawner**
This adds QUALITY_DECLARATION files for rmw_fastrtps_cpp and rmw_fastrtps_shared_cpp and package-level READMEs.

Signed-off-by: Stephen Brawner <brawner@gmail.com>

### CODE REVIEW

File: rmw_fastrtps_cpp/QUALITY_DECLARATION.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+Some of the `rmw_fastrtps_cpp` features are documented on the repository level [README](../README.md).
+Much of Fast RTPS itself has feature documentation [hosted publicly](https://fast-rtps.docs.eprosima.com/en/latest).
+
+### Public API Documentation [3.ii]
+
+Most of `rmw_fastrtps_cpp` has embedded API documentation. It is not yet hosted publicly.
+
+### License [3.iii]
+
+The license for `rmw_fastrtps_cpp` is Apache 2.0, and a summary is in each source file, the type is declared in the `package.xml` manifest file, and a full copy of the license is in the [LICENSE](../LICENSE) file.
```
Comment (ahcorde) [inline rmw_fastrtps_cpp/QUALITY_DECLARATION.md:70]:
```suggestion
The license for `rmw_fastrtps_cpp` is Apache 2.0, and a summary is in each source file, the type is declared in the [package.xml](./package.xml) manifest file, and a full copy of the license is in the [LICENSE](../LICENSE) file.
```

File: rmw_fastrtps_shared_cpp/QUALITY_DECLARATION.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+Some of the `rmw_fastrtps_shared_cpp` features are documented on the repository level [README](../README.md).
+Much of Fast RTPS itself has feature documentation [hosted publicly](https://fast-rtps.docs.eprosima.com/en/latest).
+
+### Public API Documentation [3.ii]
+
+Most of `rmw_fastrtps_shared_cpp` has embedded API documentation. It is not yet hosted publicly.
+
+### License [3.iii]
+
+The license for `rmw_fastrtps_shared_cpp` is Apache 2.0, and a summary is in each source file, the type is declared in the `package.xml` manifest file, and a full copy of the license is in the [LICENSE](../LICENSE) file.
```
Comment (ahcorde) [inline rmw_fastrtps_shared_cpp/QUALITY_DECLARATION.md:70]:
```suggestion
The license for `rmw_fastrtps_shared_cpp` is Apache 2.0, and a summary is in each source file, the type is declared in the [package.xml](./package.xml) manifest file, and a full copy of the license is in the [LICENSE](../LICENSE) file.
```

File: rmw_fastrtps_cpp/QUALITY_DECLARATION.md (hunk 1 above)
Comment (brawner) [reply rmw_fastrtps_cpp/QUALITY_DECLARATION.md]: Done.

File: rmw_fastrtps_shared_cpp/QUALITY_DECLARATION.md (hunk 2 above)
Comment (brawner) [reply rmw_fastrtps_shared_cpp/QUALITY_DECLARATION.md]: Done.

File: rmw_fastrtps_shared_cpp/QUALITY_DECLARATION.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+## Testing [4]
+
+### Feature Testing [4.i]
+
+There is no feature testing in `rmw_fastrtps_shared_cpp`.
+
+### Public API Testing [4.ii]
+
+There is little API testing in `rmw_fastrtps_shared_cpp`.
```
Comment (ahcorde) [inline rmw_fastrtps_shared_cpp/QUALITY_DECLARATION.md:92]: add link https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rmw_fastrtps_shared_cpp/DDSAttributesToRMWQosTest/

Comment (brawner) [reply rmw_fastrtps_shared_cpp/QUALITY_DECLARATION.md]: Added.

**Pull request merged**